### ISSUE TITLE
Opening delimiters preceding letters like »f« or »y«

### ISSUE THREAD

**Issue by JakobWerner** [YuanshengZhao/Garamond-Math#3]
\documentclass{article}
	
	\usepackage{unicode-math}
	\setmathfont{Garamond-Math}
	
	\begin{document}
		\( (f) \quad \{f\} \quad [f] \quad \lvert f\rvert \)
		
		\( (y) \quad \{y\} \quad [y] \quad \lvert y\rvert \)
	\end{document}

![screenshot from 2018-05-08 09-36-29](https://user-images.githubusercontent.com/37799533/39744117-6cdcd132-52a3-11e8-8bcb-086745545e75.png)

I think those will need some treatment. I don't know what will be an appropriate approach however. I'm not sure kerning can solve the issue without making everything look asymmetric. In a text font this looks like maybe ligatures are needed, but this is perhaps not the correct thing to do for a math font. So maybe the glyphs of the letters need to be adjusted for the math font?

**Comment by YuanshengZhao** [comment]
![untitled](https://user-images.githubusercontent.com/37881856/39768048-d2f5b57a-531a-11e8-8848-cc6bc589f51b.png)
![untitled1](https://user-images.githubusercontent.com/37881856/39768061-d8e9ca2a-531a-11e8-90a9-77222bffcad4.png)

These are the effects (the first two rows are XITS), rough though. In fact the height of "f" is larger than the parenthesis, it might be better to use larger parenthesis, and the second picture may look better.

**Comment by Stone-Zeng** [comment]
What about `\mathit` and `\symit`? @yuanshengzhao

**Comment by svenper** [comment]
What about changing the sidebearings of the letters? E.g. Computer Modern does this, which makes character spacing inconsistent (which should be irrelevant for a math font, where letters are variables, not parts of words).

**Comment by YuanshengZhao** [comment]
Thank you very much for your reply! In fact, this is part of what is done. Changing the sidebearings of letter certainly an essential part. Besides, the sidebearings of ()[]||, etc, is also widened. I do not think the metric has been adjusted very well and more adjustment will be made when new problems are found.

BTW, I admit that character spacing is not so important in math, sometimes too inconsistent spacing may result in a equation not look very nice. Also, there is the case that `$SO(3)$` will be used. So, some concession is needed.

![image](https://user-images.githubusercontent.com/37881856/44439989-11496180-a601-11e8-991d-e4db71ae3071.png)

**Comment by JakobWerner** [comment]
I don't think, $SO(3)$ should ever be used. It should always be $\operatorname{SO}(3)$ or something like this, so that `SO` will be printed in the text font.

**Comment by Stone-Zeng** [comment]
Upright and italic `SO` are both widely used, and it's more like a preference.

Personally, I will use `\mathit{SO}(3)`, or

```tex
\def\SO{\mathit{SO}}
\SO(3)
```

**Comment by YuanshengZhao** [comment]
Another comment here. In the latest version, the spacing of bracket has been adjusted to a (maybe, and only works on XeTeX) acceptable value. But the problem persist on `VA`, `V_A`, etc, where the spacing is just too large. 

In math, no kerning on `VA`, this is fine in most cases, but as Garamond has large italic angle, the spacing is too large, even if it is regard as product between `V` and `A`. For subscripts, as XeTeX and LuaTeX with unicode-math have somewhat strange behavior to `MathKerning` (XeTeX only works for single char, while LuaTeX only works for multiple char), the `MathKerning` cannot be efficiently implemented (if there are any).

At present, some manual adjustment is required for these pairs.

**Comment by timtro** [comment]
> Upright and italic `SO` are both widely used, and it's more like a preference.
> 
> Personally, I will use `\mathit{SO}(3)`, or
> 
> ```
> \def\SO{\mathit{SO}}
> \SO(3)
> ```

Of course, you're right about personal preference, and I do fight my private reflexes to persuade people too emphatically. I'm also aware that this isn't the venue for discussions on style and usage. So please forgive me for self-indulgently pointing out that there may be good reasons to set SO in upright. A practical convention (supported also in the ISO 800000:2009 standard) is to set natural (logically fundamental) objects upright. We do it already with numbers, trigonometric and exponential functions and the logical extension means π, i, e, and, to the point, fundamental symmetry groups like SO. (Note, that I'm not suggesting that mathematical literature should be held to a standard: that would stifle quite a lot of important expression.)

YuanshengZhao---If you would like me to delete this comment, let me know.
